File: 프로그래머스/lv1_같은_숫자는_싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;

    for(int i=0; i<arr.size(); i++){
        if( i==0 ){
            answer.push_back(arr[i]);
        }
        else if(answer[answer.size()-1] != arr[i]){
            answer.push_back(arr[i]);
        }
    }

    return answer;
}

int main(void){
    vector<int> arr = {4,4,4,3,3};
    vector<int> answer;

    answer = solution(arr);
    
    for( auto elem : answer){
        cout << elem;
    }

    return 0;
}

File: 프로그래머스/lv1_같은_숫자는_싫어.py
def solution(arr):
    answer = []
    
    for i in arr:
        if len(answer) == 0:
            answer.append(i)
        else:
            if(answer[-1] != i):
                answer.append(i)


    return answer


if __name__ == "__main__":
    arr = [1,1,3,3,0,1,1]
    ret = solution(arr)
    print(ret)